### PR TITLE
chore: convert kml module to typescript

### PR DESCRIPTION
**Pull request by morphoix**

### CODE REVIEW

File: modules/kml/test/kml-loader.spec.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import test from 'tape-promise/tape';
 import {validateLoader} from 'test/common/conformance';
 
-import {fetchFile, load} from '@loaders.gl/core';
+import {fetchFile, load} from '@loaders.gl/core/';
```
Comment (dryabinin-actionengine) [inline modules/kml/test/kml-loader.spec.js]: Please remove `/`

**Review by dryabinin-actionengine — APPROVED** [review]

File: modules/kml/test/kml-loader.spec.js [hunk 2]
```diff
@@ -24,13 +24,13 @@ test('KMLLoader#loader conformance', (t) => {
...
   const KML = await response.text();
-
-  let isKML = KML.startsWith(KMLLoader.tests[0]);
-  t.equal(isKML, true, 'Correctly accepted valid KML');
-
-  isKML = INVALID_KML.startsWith(KMLLoader.tests[0]);
-  t.equal(isKML, false, 'Correctly rejected invalid KML');
-
+  const a = (KMLLoader.tests && KMLLoader.tests[0]) || '';
+  if (typeof a === 'string') {
```
Comment (kylebarron) [inline modules/kml/test/kml-loader.spec.js:28]: Why do you need this `typeof`? For one, when will `KMLLoader.tests[0]` ever not exist? And two, if that doesn't exist, then `a` will be `''`, so this if clause will always run.

Can I suggest also naming the variable better than just `a`? Maybe

```js
const KMLTest = KMLLoader.tests && KMLLoader.tests[0]
if (KMLTest) {
  // check test
}
```

File: modules/kml/src/gpx-loader.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import {geojsonToBinary} from '@loaders.gl/gis';
+import {LoaderObject} from '@loaders.gl/loader-utils/types';
```
Comment (ibgreen) [inline modules/kml/src/gpx-loader.ts:2]: Use `import type {` when importing pure types (among other things this avoids problems when we transpile to js with babel).

File: modules/kml/src/gpx-loader.ts [hunk 4]
```diff
@@ -15,7 +16,7 @@ const GPX_HEADER = `\
  * Loader for GPX (GPS exchange format)
  * @type {LoaderObject}
  */
-export const GPXLoader = {
+export const GPXLoader: LoaderObject = {
```
Comment (ibgreen) [inline modules/kml/src/gpx-loader.ts:19]: Drop the jsdoc `@type`. You can seach the entire module for `@type` those should be removed in TS files.

**Review by ibgreen — APPROVED** [review]

File: modules/kml/test/kml-loader.spec.js (hunk 2 above)
Comment (morphoix) [reply modules/kml/test/kml-loader.spec.js:28]: > Why do you need this `typeof`? For one, when will `KMLLoader.tests[0]` ever not exist? And two, if that doesn't exist, then `a` will be `''`, so this if clause will always run.

I inserted it here as the variable can be a different type, callback or array buffer, not only string

Comment (morphoix) [reply modules/kml/test/kml-loader.spec.js:28]: > Can I suggest also naming the variable better than just `a`? Maybe

Changed to 'KMLTest', thanks for your notice :)

**Pull request merged**